server/rosetta/cosmos/conversion/data.go

```diff
@@ -0,0 +1,297 @@
+		}
+	}
+	return converted
+}
+
+// SdkTxToOperations converts a tx response to operations
+func SdkTxToOperations(tx sdk.Tx, withStatus bool) []*types.Operation {
+	var operations []*types.Operation
+
+	feeOps := GetFeeOperationsFromTx(tx, withStatus)
+	operations = append(operations, feeOps...)
+
+	msgOps := SdkMsgsToOperations(tx.GetMsgs(), withStatus, len(feeOps))
+	operations = append(operations, msgOps...)
+
+	return operations
+}
+
+func GetFeeOperationsFromTx(tx sdk.Tx, withStatus bool) []*types.Operation {
+	verifiableTx := tx.(sdk.FeeTx)
+	feeCoins := verifiableTx.GetFee()
+	var ops []*types.Operation
+	if feeCoins != nil {
+		var feeOps = GetFeeOpFromCoins(feeCoins, verifiableTx.FeePayer().String(), withStatus)
+		ops = append(ops, feeOps...)
+	}
+
+	return ops
+}
+
+// TendermintTxsToTxIdentifiers converts a tendermint raw transaction into a rosetta tx identifier
+func TendermintTxsToTxIdentifiers(txs []tmtypes.Tx) []*types.TransactionIdentifier {
+	converted := make([]*types.TransactionIdentifier, len(txs))
+	for i, tx := range txs {
+		converted[i] = &types.TransactionIdentifier{Hash: fmt.Sprintf("%x", tx.Hash())} // TODO hash is sha256, so we hex it?
+	}
+	return converted
+}
+
+// TendermintBlockToBlockIdentifier converts a tendermint result block to a rosetta block identifier
+func TendermintBlockToBlockIdentifier(block *tmcoretypes.ResultBlock) *types.BlockIdentifier {
+	return &types.BlockIdentifier{
+		Index: block.Block.Height,
+		Hash:  block.Block.Hash().String(),
+	}
+}
+
+func SdkMsgsToOperations(msgs []sdk.Msg, withStatus bool, feeLen int) []*types.Operation {
+	var operations []*types.Operation
+	var status string
+	for i, msg := range msgs {
+		switch msg.Type() { // nolint
+		case rosetta.OperationSend:
+			newMsg := msg.(*banktypes.MsgSend)
+			fromAddress := newMsg.FromAddress
+			toAddress := newMsg.ToAddress
+			amounts := newMsg.Amount
+			if len(amounts) == 0 {
+				continue
+			}
+			coin := amounts[0]
+			sendOp := func(account, amount string, index int) *types.Operation {
+				if withStatus {
+					status = rosetta.StatusSuccess
+				}
+				return &types.Operation{
+					OperationIdentifier: &types.OperationIdentifier{
+						Index: int64(index),
+					},
+					Type:   rosetta.OperationSend,
+					Status: status,
+					Account: &types.AccountIdentifier{
+						Address: account,
+					},
+					Amount: &types.Amount{
+						Value: amount,
+						Currency: &types.Currency{
+							Symbol: coin.Denom,
+						},
+					},
+				}
+			}
+			operations = append(operations,
+				sendOp(fromAddress, "-"+coin.Amount.String(), feeLen+i),
+				sendOp(toAddress, coin.Amount.String(), feeLen+i+1),
+			)
+		}
+	}
+	return operations
+}
+
+func GetMsgsFromOperations(ops []*types.Operation) (sdk.Msg, sdk.Coins, error) {
+	var feeAmnt []*types.Amount
+	var sendOps []*types.Operation
+	if len(ops) == 2 {
+		sendMsg, err := GetTransferTxDataFromOperations(ops)
+		return sendMsg, nil, err
+	} else if len(ops) == 3 {
```

Comment:
You don't need the else clause because the `if` branch ends with a `return` statement.

```suggestion
	}

	if len(ops) == 3 {
```